packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -725,8 +727,12 @@ export class SnapController extends BaseController<
         ...{
           ...state,
           snaps: Object.values(state?.snaps ?? {}).reduce(
-            (memo: Record<string, Snap>, snap) => {
+            (
+              memo: Record<ValidatedSnapId, Snap>,
+              snap: Snap & { sourceCode: string },
+            ) => {
               // eslint-disable-next-line @typescript-eslint/no-unused-vars
+              // this is to catch any old state that may still have sourceCode in the snap state
```

Comment:
Not sure I would phrase it this way. We strip it out to prevent piping it to the MetaMask UI. We store it in the runtime while we are running and then persist it to state when needed.